src/Language/PureScript/AST/Declarations.hs

```diff
@@ -154,7 +154,7 @@ data Declaration
   -- A type instance declaration (name, dependencies, class name, instance types, member
   -- declarations)
   --
-  | TypeInstanceDeclaration Ident [Constraint] (Qualified ProperName) [Type] [Declaration]
+  | TypeInstanceDeclaration Ident [Constraint] (Qualified ProperName) [Type] (Maybe [Declaration])
```

Comment:
Another option here would be a separate term in the sum which encodes derived instances, with the advantages that (a) you don't force all the code to handle the `Nothing` case, and (b) it's arguably another constructor may be clearer to developers than an optional `Declaration`.